bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -68,6 +71,14 @@ public void addScriptEngineFactory(ScriptEngineFactory provider) {
         for (String language : provider.getLanguages()) {
             this.supportedLanguages.put(language, provider);
         }
+
+        TreeMap<String, String> languageNames = provider.getLanguageNames();
+
+        for (Map.Entry<String, String> entry : languageNames.entrySet()) {
+            if (!this.supportedLanguageNames.containsKey(entry.getKey())) {
```

Comment:
Use [computeIfAbsent](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-)

Reply:
This was removed.